### PR TITLE
[ci] add github action to build 1.3 firmware

### PR DESCRIPTION
**Pull request by sunytt**
Added a github action to build 1.3 nrf52840 firmware and save for a day.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          cd /tmp
+          sudo apt-get update
+          sudo apt-get --no-install-recommends install -y ninja-build gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf
+          wget --tries 4 --no-check-certificate --quiet https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2019q4/RC2.1/gcc-arm-none-eabi-9-2019-q4-major-x86_64-linux.tar.bz2 -O gcc-arm.tar.bz2
+          tar xjf gcc-arm.tar.bz2
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          submodules: recursive
+      - name: build reference release for 1.3
```
Comment (superwhd) [inline .github/workflows/build.yml:51]:
```suggestion
      - name: Build reference release for 1.3
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        run: |
+          export PATH=/tmp/gcc-arm-none-eabi-9-2019-q4-major/bin:$PATH
+          REFERENCE_PLATFORM=nrf52840 REFERENCE_RELEASE_TYPE=1.3  ./script/make-reference-release.bash
+      - name: save reference releases
```
Comment (superwhd) [inline .github/workflows/build.yml:55]: Maybe we can use the default name of this step to be consistent with OT.
https://github.com/openthread/openthread/blob/main/.github/workflows/otbr.yml#L198

File: .github/workflows/build.yml (hunk 1 above)
Comment (sunytt) [reply .github/workflows/build.yml]: Done

File: .github/workflows/build.yml (hunk 2 above)
Comment (sunytt) [reply .github/workflows/build.yml]: Done

**Review by librasungirl — APPROVED** [review]

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**